File: hssimpleconc/simpleconc2.hs
module Main where

import Control.Concurrent (forkIO, threadDelay)
import Control.Concurrent.MVar (newEmptyMVar, takeMVar, putMVar)

main :: IO ()
main = do
    result <- newEmptyMVar

    _ <- forkIO (do
        sleep 5
        putStrLn "Calculated result!"
        putMVar result 42)

    putStrLn "Waiting..."
    value <- takeMVar result
    putStrLn ("The answer is: " ++ show value)

sleep :: Int -> IO ()
sleep n = threadDelay (n * 1000)

File: hssimpleconc/simpleconc2.go
package main

import (
	"fmt"
	"time"
)

func main() {
	c := make(chan string)

	go func() {
		time.Sleep(time.Millisecond * 5)
		fmt.Println("Calculated result!")
		c <- "42"
	}()

	fmt.Println("Waiting...")
	fmt.Println("The answer is: " + <-c)
}
